File: packages/authentication/ios/Plugin/Classes/GetIdTokenResultResult.swift
import Foundation
import FirebaseAuth
import Capacitor

@objc class GetIdTokenResultResult: NSObject {
    let tokenResult: AuthTokenResult?

    init(tokenResult: AuthTokenResult?) {
        self.tokenResult = tokenResult
    }

    func toJSObject() -> JSObject {
        var result = JSObject()

        if let tokenResult = tokenResult {
            // Convert to milliseconds (JS commonly expects ms since epoch)
            result["authTime"] = Int(tokenResult.authDate.timeIntervalSince1970 * 1000)
            result["expirationTime"] = Int(tokenResult.expirationDate.timeIntervalSince1970 * 1000)
            result["issuedAtTime"] = Int(tokenResult.issuedAtDate.timeIntervalSince1970 * 1000)
            result["signInProvider"] = tokenResult.signInProvider
            result["signInSecondFactor"] = tokenResult.signInSecondFactor

            var claims = JSObject()
            for (key, value) in tokenResult.claims {
                if let jsonValue = value as? JSValue {
                    claims[key] = jsonValue
                }
            }
            result["claims"] = claims
        }

        return result
    }
}

File: packages/authentication/android/src/main/java/app/capgo/capacitor/firebase/authentication/GetIdTokenResultResult.java
package app.capgo.capacitor.firebase.authentication.classes;

import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import com.google.firebase.auth.GetTokenResult;
import app.capgo.capacitor.firebase.authentication.interfaces.Result;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.TimeZone;

public class GetIdTokenResultResult implements Result {

    @Nullable
    private GetTokenResult tokenResult;

    private SimpleDateFormat iso8601Format;

    public GetIdTokenResultResult(@Nullable GetTokenResult tokenResult) {
        this.tokenResult = tokenResult;
    }

    public JSObject toJSObject() {
        JSObject result = new JSObject();
        if (tokenResult != null) {
            result.put("authTime", tokenResult.getAuthTimestamp() * 1000L);
            result.put("expirationTime", tokenResult.getExpirationTimestamp() * 1000L);
            result.put("issuedAtTime", tokenResult.getIssuedAtTimestamp() * 1000L);
            result.put("signInProvider", tokenResult.getSignInProvider());
            result.put("signInSecondFactor", tokenResult.getSignInSecondFactor());

            JSObject claims = new JSObject();
            for (String key : tokenResult.getClaims().keySet()) {
                claims.put(key, tokenResult.getClaims().get(key));
            }
            result.put("claims", claims);
        }
        return result;
    }
}
